src/Browser.php

```diff
@@ -335,4 +337,64 @@ public function withOptions(array $options)
 
         return $browser;
     }
+
+    public function withTimeout($timeout)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'timeout' => $timeout,
+        ));
+
+        return $browser;
+    }
+
+    public function withoutTimeout()
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'timeout' => null,
+        ));
+
+        return $browser;
+    }
+
+    public function withFollowRedirects($followRedirects)
```

Comment:
`withFollowRedirects()` LGTM, but not sure about `withMaxRedirects()`.

Perhaps it makes sense to merge both into a single method and trigger this logic based on the value given (`bool|int`)?

Reply:
Doesn't feel right to do that tbh. They are both closely related but distinctly different

Reply:
I'm undecided. Can you give a use case to explicitly control both parameters individually?

It's currently my understanding this could be better covered by a single method accepting different parameter types instead.

Reply:
What about dropping `withFollowRedirects` altogether, both here and deeper inside the package? `withMaxRedirects` already implies we'll follow redirects, setting that to `0` will be the same as `withFollowRedirects(false)` anyway.

Reply:
I guess we need some better documentation :-)

They're somewhat similar, but they actually have distinct behavior: setting to `followRedirects=false` means we do not *follow* redirects, i.e. the request may be fulfilled with a response with status code 3xx (and `maxRedirects` has no effect here). Setting `maxRedirects=0` means we do not *accept* redirects, i.e. the request will be rejected when a redirect is encountered (only if `followRedirects=true` is set).

I agree that this behavior is confusing, hence my suggestion to merge both into a single method :+1:

Reply:
To be honest that would suggest they should be two different methods rather than combining them into one to me ;).